src/Shapes.hs

```diff
@@ -809,6 +828,25 @@ lineTransformType = ("LineTransform", M.fromList
         ("name", (StrT, constValue $ StrV "defaultLine"))
     ])
 
+squareTransformType = ("SquareTransform", M.fromList
+    [
+        -- knowing that the "original" is centered at origin, still need to store?
+        ("x", (FloatT, x_sampler)), -- x and y as dx and dy in transform?
+        ("y", (FloatT, y_sampler)),
+        ("side", (FloatT, width_sampler)), -- as both scaleX and scaleY ? 
+        ("rotation", (FloatT, constValue $ FloatV 0.0)), -- initialize as 0?
+        -- matrices
+        ("transform", (FloatT, constValue $ HMatrixV idH)),
+        ("transformation", (FloatT, constValue $ HMatrixV idH)),
+        -- ones that remain the same
+        ("color", (ColorT, sampleColor)),
+        ("style", (StrT, constValue $ StrV "none")), -- TODO: what is this?
+        ("strokeColor", (ColorT, sampleColor)),
+        ("strokeWidth", (FloatT, constValue $ FloatV 0.0)),
+        ("name", (StrT, constValue $ StrV "defaultSquare")),
+        ("polygon", (PtListT, constValue $ PtListV [])) -- Q: why empty?
```

Comment:
I just arbitrarily initialize the polygon to be nothing since the polygon property gets computed by the function later